src/tdyio.c

```diff
@@ -41,16 +45,57 @@ PetscErrorCode TDyIOWriteVec(TDy tdy){
     TDyIOPrintVec(v, time);
   }
   if (tdy->io->format == ExodusFormat){
+    char *ofilename = tdy->io->exodus_filename;
```

Comment:
If we are handling both formats at the same time, these conditionals are fine, but otherwise, should be a if/else.